util/journaldb/src/traits.rs

```diff
@@ -21,13 +21,28 @@ use std::sync::Arc;
 
 use bytes::Bytes;
 use ethereum_types::H256;
-use hashdb::HashDB;
+use hash_db::{HashDB, AsHashDB};
 use keccak_hasher::KeccakHasher;
 use kvdb::{self, DBTransaction, DBValue};
+use std::collections::HashMap;
+
+
+/// expose keys of a hashDB for debugging or tests (slow).
+pub trait KeyedHashDB: HashDB<KeccakHasher, DBValue> {
+  /// Primarily use for tests, highly inefficient.
```

Comment:
indentation
```suggestion
	/// Primarily use for tests, highly inefficient.
```

Reply:
If `keys()` is only used for tests, maybe put it behind `#[cfg(test)]`?

Reply:
It is actually use on non purely test context (eg with evmbin for debug display of trie state), it could still be put behind a feature.

Reply:
I couldn't find any uses of `keys()` besides tests, could you point me to the code?

Reply:
I may be mistaking (maybe it is simply triedb iterator that is used, I will test the change in a bit).

Reply:
So  you are right :+1: . I can build without it and it seems fine, but I am having a hard time making the test run, I believe it is related to the fact that we use it in other crates (having conditional trait method seems ok for `commit_batch` or `inject_batch`, but those are only use in journaldb). The same thing happens when you depend on a crate and expect the test function to be here (thus the 'test_helper' crates).

Reply:
Let's merge the PR then. I'll open an issue to make `KeyedHashDB` test-only.